ecommerce/models.py

```diff
@@ -291,6 +288,22 @@ def check_validity(self, user: User):
 
         return self.valid_now()
 
+    def check_validity_with_products(self, products: list):
+        """
+        Enforces the redemption rules for a given discount.
+
+        Args:
+            - products (list): List of products.
+        Returns:
+            - boolean
```

Comment:
The docstrings need to be standardized for all the functions that you added. An example of that can be seen [here](https://github.com/mitodl/mitxonline/blob/main/courses/api.py#L128). This will apply to other functions in this PR too.

Reply:
I just followed [this](https://github.com/mitodl/mitxonline/blob/421f2b9d3707cd72d135f0fa8ae375f231780c8b/ecommerce/models.py#L261) one docstring format. I'll change it according to the format you mentioned

Reply:
It should be standardized everywhere. Like in this PR different functions had different formats.